library/lambda_visitor.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <boost/variant.hpp>
+
+template <typename ReturnType, typename... Lambdas>
+struct lambda_visitor;
+
+template <typename ReturnType, typename Lambda1, typename... Lambdas>
+struct lambda_visitor< ReturnType, Lambda1 , Lambdas...>
+: public lambda_visitor<ReturnType, Lambdas...>
+, public Lambda1 {
+	using Lambda1::operator();
+	using lambda_visitor<ReturnType, Lambdas...>::operator();
+
+	explicit lambda_visitor(Lambda1 l1, Lambdas... lambdas)
+	: lambda_visitor<ReturnType, Lambdas...> (lambdas...)
+	, Lambda1(l1)
+	{}
+};
+
+
+template <typename ReturnType, typename Lambda1>
+struct lambda_visitor<ReturnType, Lambda1>
+: public boost::static_visitor<ReturnType>, public Lambda1 {
+	using Lambda1::operator();
+
+	explicit lambda_visitor(Lambda1 l1)
+	: boost::static_visitor<ReturnType>()
+	, Lambda1(l1)
+	{}
+};
+
+template <typename ReturnType>
+struct lambda_visitor<ReturnType>
```

Comment:
Seems useless, end-of-recursion is above (`lambda_visitor<ReturnType, Lambda1>`). Doesn't code compile without it? I hope does.